api/app/GraphQL/Mutations/PublishPoolAdvertisement.php

```diff
@@ -21,9 +19,9 @@ public function __invoke($_, array $args)
         $poolAdvertisement = Pool::find($args['id'])
             ->load(['classifications', 'essentialSkills', 'nonessentialSkills']);
         $poolValidation = new PublishPoolAdvertisementValidator;
-        $validator = Validator::make($poolAdvertisement->toArray(), $poolValidation->rules()); // First validate pool advertisement before updating.
+        $validator = Validator::make($poolAdvertisement->toArray(), $poolValidation->rules(), $poolValidation->messages()); // First validate pool advertisement before updating.
         if ($validator->fails()) {
-            throw new ValidationException($validator);
+            throw FormattedError::setInternalErrorMessage($validator->errors()->first());
```

Comment:
I think this is causing the weird debug error:
![image](https://user-images.githubusercontent.com/8978655/182383048-c4693c08-455d-4e3b-a3f8-ef0fe016f451.png)
I think this function returns void and you can't throw that.

Reply:
Okay so I fixed this problem by using the ValidationException in [commit #4fdf634 ](https://github.com/GCTC-NTGC/gc-digital-talent/pull/3547/commits/4fdf6e42ad9ac81c6c79a1d73f75b5cbbecf02c9). However, this is a temporary fix until the Custom error messages issue gets done. The ValidationException returns all the fields that failed validation, and it would be better for the user if they can see all the inputs that failed at once. 

![image](https://user-images.githubusercontent.com/22059495/182403131-bb5ce6d4-7373-41bd-922c-fec76057a05f.png)

Reply:
I like it.  The naive implementation can just use `message` while a smarter implementation can examine the extensions.